pylint/checkers/base/basic_error_checker.py

```diff
@@ -95,6 +95,71 @@ def x(self, value): self._x = value
     return False
 
 
+def _extract_register_target(dec: nodes.NodeNG) -> nodes.NodeNG | None:
+    """
+    If decorator `dec` looks like `@func.register(...)` or `@func.register`,
+    return the `func` target node (Name or Attribute). Otherwise return None.
+    """
+    if isinstance(dec, nodes.Call):
+        func_part = dec.func
+        if isinstance(func_part, nodes.Attribute) and func_part.attrname == "register":
+            return func_part.expr
+        return None
+
+    if isinstance(dec, nodes.Attribute) and dec.attrname == "register":
+        return dec.expr
+
+    return None
```

Comment:
I've been contaminated by Marc and I see them everywhere now, sorry.

```suggestion
def _extract_register_target(decorator: nodes.NodeNG) -> nodes.NodeNG | None:
    match decorator:
        case nodes.Call(func=nodes.Attribute(attrname="register", expr=expr)):
            # Looks like `@func.register(...)`
            return expr
        case nodes.Attribute(attrname="register", expr=expr):
            # Looks like `@func.register`
            return expr
        case _:
            return None
```

Reply:
Thanks! The `match` version looks elegant. I will use it in future refactors.